src/driver/xdma_sgdma.c

```diff
@@ -261,22 +263,51 @@ ssize_t xdma_channel_read_write(struct xdma_channel *chnl,
 		return -EINVAL;
 	}
 
-	rv = check_transfer_align(engine, buf, count, *pos, 1);
-	if (rv) {
-		pr_info("Invalid transfer alignment detected\n");
-		return rv;
-	}
+	if (sg_alloc_table(&sgt, SG_MAX_SINGLE_ALLOC, GFP_KERNEL))
+		return -ENOMEM;
+
+	while (max_bytes && iov_iter_count(iter)) {
+		pages = (struct page **) sgt.sgl;
+		pages += SG_MAX_SINGLE_ALLOC * (PAGE_PTRS_PER_SGL - 1);
+
+		size = iov_iter_get_pages(iter, pages, max_bytes, SG_MAX_SINGLE_ALLOC, &offset);
+		if (size < 0)
+			return size;
```

Comment:
Need to call sg_free_table() before returning here and in a number of places below.

Reply:
Ops, yes